com.unity.netcode.gameobjects/Editor/CodeGen/NetworkBehaviourILPP.cs

```diff
@@ -934,6 +934,19 @@ private void InjectWriteAndCallBlocks(MethodDefinition methodDefinition, CustomA
                 {
                     var paramDef = methodDefinition.Parameters[paramIndex];
                     var paramType = paramDef.ParameterType;
+                    if (paramType.FullName == CodeGenHelpers.ClientRpcSendParams_FullName ||
+                        paramType.FullName == CodeGenHelpers.ClientRpcReceiveParams_FullName)
+                    {
+                        m_Diagnostics.AddError($"Rpcs may not accept {paramType.FullName} as a parameter. Use {nameof(ClientRpcParams)} instead.");
+                        continue;
+                    }
+
+                    if (paramType.FullName == CodeGenHelpers.ServerRpcSendParams_FullName ||
+                        paramType.FullName == CodeGenHelpers.ServerRpcReceiveParams_FullName)
+                    {
+                        m_Diagnostics.AddError($"Rpcs may not accept {paramType.FullName} as a parameter. Use {nameof(ServerRpcParams)} instead.");
+                        continue;
+                    }
```

Comment:
I guess what's going on here is that even though `ClientRpcSendParams` and `ServerRpcSendParams` are not `INetworkSerializable` types, they are still being serialized with `.WriteValue(struct)` API and that's unintentional & confusing, right?

Reply:
Yes, that's correct. Anything that's an unmanaged struct can be sent as an RPC parameter automatically and in this case that creates confusion since *RpcParams has special behavior.